### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by AxelSoler**
## Working with pgAdmin4 app (NEW SCREENSHOTS ARE AT THE END OF THE COMMENTS SECTION)

- Add a column species of type string to your animals table.
- Insert animals data Captura de pantalla 2022-07-05 124858
- Inside a transaction update the animals table by setting the species column to unspecified.
Captura de pantalla 2022-07-05 132955

- Then roll back the change
Captura de pantalla 2022-07-05 133221

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Captura de pantalla 2022-07-05 133810

- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Captura de pantalla 2022-07-05 134033

- Commit the transaction.
- Inside a transaction delete all records in the animals table, then roll back the transaction.
Captura de pantalla 2022-07-05 135135
Captura de pantalla 2022-07-05 135217

- Delete all animals born after Jan 1st, 2022.
Captura de pantalla 2022-07-05 135514

- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
Captura de pantalla 2022-07-05 141350

- Rollback to the savepoint and Update to fix initial problem of values
Captura de pantalla 2022-07-05 142032

- Update all animals' weights that are negative to be their weight multiplied by -1.
Captura de pantalla 2022-07-05 142217

- Commit transaction

## Write queries to answer the following questions:
How many animals are there?
Captura de pantalla 2022-07-05 142653

How many animals have never tried to escape?
Captura de pantalla 2022-07-05 142838

What is the average weight of animals?
Captura de pantalla 2022-07-05 143005

Who escapes the most, neutered or not neutered animals?
Captura de pantalla 2022-07-05 144031

What is the minimum and maximum weight of each type of animal?
Captura de pantalla 2022-07-05 144426

What is the average number of escape attempts per animal type of those born between 1990 and 2000
Captura de pantalla 2022-07-05 144835
?

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -24,3 +24,80 @@ SELECT name FROM animals WHERE name != 'Gabumon'
...
+
+-- Inside a transaction delete all records in the animals table, then roll back the transaction.
+BEGIN
+
+DELETE FROM animals
+ROLLBACK
+
+-- Delete all animals born after Jan 1st, 2022.
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
```
Comment (CollinsTatang) [inline queries.sql]: - [x] Kindly correct your query to
```DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;```

File: queries.sql [hunk 2]
```diff
@@ -24,3 +24,80 @@ SELECT name FROM animals WHERE name != 'Gabumon'
...
+SELECT neutered FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT (MAX(weight_kg), MIN(weight_kg), species) FROM animals GROUP BY species
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts), species FROM animals
+WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-01-01'
+GROUP BY species
```
Comment (CollinsTatang) [inline queries.sql]: - [x]  Kindly correct your query to
```sql
SELECT species, AVG(escape_attempts) FROM animals
   GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @AxelSoler, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
Please update your screenshot query to match the requested instruction for the following

- [ ] Insert animals data.
- [ ] Inside a transaction update the animals table by setting the species column to unspecified.
- [ ] Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
- [ ] Inside a transaction delete all records in the animals table, then roll back the transaction.
- [ ] Delete all animals born after Jan 1st, 2022.
- [ ] Update all animals' weight to be their weight multiplied by -1.
- [ ] Rollback to the savepoint and Update to fix initial problem of values
You currently have just one query `SELECT  *  FROM animals;`


_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AxelSoler** [comment]
Hello @CollinsTatang, actually I don't understand what you mean for the screenshots to match the requested instruction. All the screenshots are showing the current step in the activity and are displayed in the table with SELECT * FROM animals;
Kindly look that the screenshots are showing the result of each activity

**Comment by CollinsTatang** [comment]
> Hello @CollinsTatang, actually I don't understand what you mean for the screenshots to match the requested instruction. All the screenshots are showing the current step in the activity and are displayed in the table with SELECT * FROM animals; Kindly look that the screenshots are showing the result of each activity

Hi @AxelSoler 
Look at the image below instead to have an UPDATE query for an update case you have SELECT * FROM animals;
query

**Comment by AxelSoler** [comment]
hello @CollinsTatang, that is because I have to SELECT ALL in order to see the table, and I did the UPDATE to reach that

**Comment by CollinsTatang** [comment]
> hello @CollinsTatang, that is because I have to SELECT ALL in order to see the table, and I did the UPDATE to reach that

Can we have a call?

**Comment by AxelSoler** [comment]
@CollinsTatang  yes

**Comment by AxelSoler** [comment]
@CollinsTatang https://microverse.zoom.us/j/94779514096?pwd=a3p0UUthZjhDeHFISXJqK3hBVGVTQT09

**Comment by CollinsTatang** [comment]
[> @CollinsTatang yes

Join Zoom Meeting
https://microverse.zoom.us/j/97581893804?pwd=NUNDMFdxWkp5UDIvc2hoaEpwbWJiZz09

Meeting ID: 975 8189 3804
Passcode: 239230
One tap mobile
+16469313860,,97581893804#,,,,*239230# US
+16694449171,,97581893804#,,,,*239230# US

Dial by your location
        +1 646 931 3860 US
        +1 669 444 9171 US
        +1 669 900 6833 US (San Jose)
        +1 929 205 6099 US (New York)
        +1 253 215 8782 US (Tacoma)
        +1 301 715 8592 US (Washington DC)
        +1 312 626 6799 US (Chicago)
        +1 346 248 7799 US (Houston)
Meeting ID: 975 8189 3804
Passcode: 239230
Find your local number: https://microverse.zoom.us/u/arOWEbxy0




](https://microverse.zoom.us/j/97581893804?pwd=NUNDMFdxWkp5UDIvc2hoaEpwbWJiZz09)

**Comment by AxelSoler** [comment]
I am having an error, it says invalid id @CollinsTatang

**Comment by CollinsTatang** [comment]
> I am having an error, it says invalid id @CollinsTatang

https://microverse.zoom.us/j/94614599497?pwd=Ly85cnJzdVJNTlliOVl0VkswS0tRUT09

**Comment by AxelSoler** [comment]
- Insert animals data.
Captura de pantalla 2022-07-05 182711

- Inside a transaction update the animals table by setting the species column to unspecified.
Captura de pantalla 2022-07-05 182815

- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Captura de pantalla 2022-07-05 182908

- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Captura de pantalla 2022-07-05 182946

- Inside a transaction delete all records in the animals table, then roll back the transaction.
Captura de pantalla 2022-07-05 184733
Captura de pantalla 2022-07-05 184757

- Delete all animals born after Jan 1st, 2022.
Captura de pantalla 2022-07-05 183638
Captura de pantalla 2022-07-05 183737


- Update all animals' weight to be their weight multiplied by -1.
Captura de pantalla 2022-07-05 183812

- Rollback to the savepoint and Update to fix initial problem of values
Captura de pantalla 2022-07-05 183843
Captura de pantalla 2022-07-05 183923
Captura de pantalla 2022-07-05 184004

**Review by descholar-ceo — APPROVED** [review]
Hi @AxelSoler , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Addressed all pointed out issues ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**